### DISCUSSION TITLE
Opt-In Block Registration for Widget Area

### DISCUSSION THREAD

**Discussion by gigitux** [woocommerce/woocommerce#53659] [WooCommerce, Blocks, and Site Editing]
Currently, all Woo Blocks available in the Site Editor are also available in the Widget Area, with only a few blocks opted out:

https://github.com/woocommerce/woocommerce/blob/48eb0030dce131947367d632b8dfecedb2d71def/plugins/woocommerce/src/Blocks/BlockTypesController.php#L473-L489

While working [on a CFE impacting the Widget Area yesterday](https://github.com/woocommerce/woocommerce/issues/53640), I noticed that a few blocks do not function properly in this context:

### Single Product Block

https://github.com/user-attachments/assets/f1cce693-55ef-4ba5-9e74-9290f6dd4b74

### Classic Cart/Checkout

https://github.com/user-attachments/assets/be797599-6d03-4321-9ea4-4bfd2868d06b

Moreover, there are few blocks that I'm not sure that it makes sense that they are available in the Widget Area, like Store Notice or Catalog Sorting.

I suggest changing the logic for registering blocks in the Widget Area from an opt-out to an opt-in approach. This would ensure that only blocks explicitly designated to work in the Widget Area are included. Explicitly specifying support for these blocks means that minimum E2E tests to cover the Widget Area are required.

What do you think?

**Comment by kmanijak** [comment]
> I suggest changing the logic for registering blocks in the Widget Area from an opt-out to an opt-in approach. This would ensure that only blocks explicitly designated to work in the Widget Area are included. Explicitly specifying support for these blocks means that minimum E2E tests to cover the Widget Area are required.
> What do you think?

Totally agree! TBH I wasn't fully aware new blocks gets exposed there and what's worse, that they may malfunction in there. Changing the logic to opt-in saves us from problems like [in the CFE](https://github.com/woocommerce/woocommerce/issues/53640). In addition, it would make block themes more appealing - unless enabled in widgets explicitly, user must switch to block theme to use them.

So strong +1 from me!

**Comment by Aljullu** [comment]
Good catch with this issue. Switching to an opt-in flow also seems like the correct approach to me. :+1:

**Comment by sunyatasattva** [comment]
Agreed! Opt-in seems the correct approach in ensuring whatever we publish is compatible with this experience. We should probably put a reminder somewhere: it's good that new blocks need to be explicitly tested with the Widgets area, but I see this as an easy thing to miss.

**Comment by gigitux** [comment]
Thanks for your feedback, folks! I created a new issue https://github.com/woocommerce/woocommerce/issues/53691. 

I'm going to close this discussion, but feel free to continue the discussion if you have any additional feedback!
